src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs

```diff
@@ -96,97 +93,67 @@ internal virtual void Update(Model model)
         /// <param name="allowedMentions">The allowed mentions for this response.</param>
         /// <param name="options">The request options for this response.</param>
         /// <param name="component">A <see cref="MessageComponent"/> to be sent with this response</param>
         /// <returns>
         ///     The sent message.
         /// </returns>
-        public abstract Task<RestFollowupMessage> FollowupAsync(Embed[] embeds = null, string text = null, bool isTTS = false, bool ephemeral = false,
-             InteractionResponseType type = InteractionResponseType.ChannelMessageWithSource,
+        public abstract Task<RestFollowupMessage> FollowupAsync(string text = null, Embed[] embeds = null, bool isTTS = false, bool ephemeral = false,
              AllowedMentions allowedMentions = null, RequestOptions options = null, MessageComponent component = null);
 
         /// <summary>
         ///     Gets the original response for this interaction.
         /// </summary>
         /// <param name="options">The request options for this async request.</param>
-        /// <returns>A <see cref="RestInteractionMessage"/> that represents the intitial response, or <see langword="null"/> if there is no response.</returns>
+        /// <returns>A <see cref="RestInteractionMessage"/> that represents the initial response.</returns>
         public Task<RestInteractionMessage> GetOriginalResponseAsync(RequestOptions options = null)
             => InteractionHelper.GetOriginalResponseAsync(this.Discord, this.Channel, this, options);
 
         /// <summary>
-        ///     Acknowledges this interaction with the <see cref="InteractionResponseType.DeferredChannelMessageWithSource"/>.
+        ///     Edits original response for this interaction.
+        /// </summary>
+        /// <param name="func">A delegate containing the properties to modify the message with.</param>
+        /// <param name="options">The request options for this async request.</param>
+        /// <returns>A <see cref="RestInteractionMessage"/> that represents the initial response.</returns>
+        public async Task<RestInteractionMessage> ModifyOriginalResponseAsync(Action<MessageProperties> func, RequestOptions options = null)
+        {
+            var model = await InteractionHelper.ModifyInteractionResponse(this.Discord, this.Token, func, options);
+            return RestInteractionMessage.Create(this.Discord, model, this.Token, this.Channel);
+        }
+
+        /// <summary>
+        ///     Acknowledges this interaction.
         /// </summary>
         /// <returns>
         ///     A task that represents the asynchronous operation of acknowledging the interaction.
         /// </returns>
-        public abstract Task AcknowledgeAsync(RequestOptions options = null);
+        public abstract Task DeferAsync(RequestOptions options = null);
```

Comment:
Could you add `AcknowledgeAsync` back as a void and add the `[Obsolete]` attribute with a message telling them to use `DeferAsync` so people updating know what to use.

Reply:
Oh, i was think about it! Ok.